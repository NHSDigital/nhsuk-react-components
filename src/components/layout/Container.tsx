import React, { FC, HTMLProps } from 'react';
import classNames from 'classnames';

interface ContainerProps extends HTMLProps<HTMLDivElement> {
  fluid?: boolean;
}

const Container: FC<ContainerProps> = ({ className, fluid, ...rest }) => (
  <div
    className={classNames(
      { 'nhsuk-width-container': !fluid },
      { 'nhsuk-width-container-fluid': fluid },
      className,
    )}
    {...rest}
  />
);

export default Container;
