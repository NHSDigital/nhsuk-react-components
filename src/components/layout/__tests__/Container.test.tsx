import React from 'react';
import { render } from '@testing-library/react';
import Container from '../Container';

describe('Container', () => {
  it('matches snapshot', () => {
    const { container } = render(<Container />);

    expect(container).toMatchSnapshot();
  });

  it('adds fluid classes', () => {
    const { container } = render(<Container fluid />);

    expect(container.querySelector('.nhsuk-width-container-fluid')).toBeTruthy();
  });
});
