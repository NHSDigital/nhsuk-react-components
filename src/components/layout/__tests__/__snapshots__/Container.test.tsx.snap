// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container matches snapshot 1`] = `
<div>
  <div
    class="nhsuk-width-container"
  />
</div>
`;
